components/src/core/components/Input/TreeSelect/TreeSelect.vue

```diff
@@ -0,0 +1,545 @@
+    const dropdownOpen = ref<Boolean>(false);
+    const isAllSelected = ref<Boolean>(false);
+    const placeholderOption = ref<OptionProp | null>(null);
+
+    const levelizeOptions = (
+      optionsArr: Option[],
+      level: number,
+      parentOptions: Option[],
+    ) => {
+      for (let option of optionsArr) {
+        option['level'] = level;
+        option['parentOptions'] = parentOptions;
+        selectedIdsObject.value[option.id] = false; //initially makes all the ids false (unselected)
+        if (option.children ? option.children.length != 0 : false) {
+          levelizeOptions(option.children, level + 1, [
+            ...parentOptions,
+            option,
+          ]);
+        }
+      }
+    };
+    levelizeOptions(optionsArr.value, 1, []);
+    console.log(optionsArr.value, 'aftr lvlzed');
+    console.log(selectedIdsObject.value, 'selectedIdsObject.value');
+
+    const selectedIdsComputed = computed(() => {
+      let selectedIds = Object.keys(selectedIdsObject.value).filter(
+        k => selectedIdsObject.value[k],
+      );
+      return selectedIds;
+    });
+
+    const selectOptionsOnCheckbox = (
+      checkboxVal: Boolean,
+      optionId: string,
+    ) => {
+      let option = optionsArr.value.find((obj: {id: string}) => {
+        return obj.id == optionId;
+      });
+
+      if (checkboxVal) {
+        addChildrenToSelectedIdsArray(option);
+        if (props.selectParentsOnChildSelection) {
+          selectParentsIfAllChildrenAreSelected(option);
+        }
+        if (isIfAllOptionsSelected()) {
+          isAllSelected.value = true;
+        }
+      } else {
+        removeChildrenFromSelectedIdsArray(option);
+        if (props.selectParentsOnChildSelection) {
+          removeParentsOfUnselectedOptionsFromSelectedIdsArray(option);
+        }
+        isAllSelected.value = false;
+      }
+      emitSelectedIds();
+    };
+
+    const emitSelectedIds = () => {
+      emit('updateSelectedIds', selectedIdsComputed.value);
+    };
+
+    const addChildrenToSelectedIdsArray = (option: Option) => {
+      if (!selectedIdsObject.value[option.id]) {
+        selectedIdsObject.value[option.id] = true;
+      }
+      if (option.children ? option.children.length != 0 : false) {
+        let children = option.children;
+        for (let child of children) {
+          addChildrenToSelectedIdsArray(child);
+        }
+      }
+    };
+
+    const selectParentsIfAllChildrenAreSelected = (option: Option) => {
+      for (let i = option.parentOptions.length - 1; i >= 0; i--) {
+        let parentOption = option.parentOptions[i];
+        let allChildrenIncluded = true;
+        for (let child of parentOption.children) {
+          if (!selectedIdsObject.value[child.id]) {
+            allChildrenIncluded = false;
+            break;
+          }
+        }
+        if (allChildrenIncluded && !selectedIdsObject.value[parentOption.id]) {
+          selectedIdsObject.value[parentOption.id] = true;
```

Comment:
Should this run recursively for parents parent. As an example, when A is checked, both B and C should be checked.

- A
> - B
> > - C